ambry-api/src/main/java/com.github.ambry/rest/RestUtils.java

```diff
@@ -551,19 +551,21 @@ public static long toSecondsPrecisionInMs(long ms) {
   /**
    * Gets the {@link GetOption} required by the request.
    * @param restRequest the representation of the request.
+   * @param defaultGetOption the {@link GetOption} to use if the {@code restRequest} doesn't have one. Can be
+   * {@code null}.
    * @return the required {@link GetOption}. Defaults to {@link GetOption#None}.
    * @throws RestServiceException if the {@link RestUtils.Headers#GET_OPTION} is present but not recognized.
    */
-  public static GetOption getGetOption(RestRequest restRequest) throws RestServiceException {
-    GetOption options = GetOption.None;
-    Map<String, Object> args = restRequest.getArgs();
-    Object value = args.get(RestUtils.Headers.GET_OPTION);
+  public static GetOption getGetOption(RestRequest restRequest, GetOption defaultGetOption)
+      throws RestServiceException {
+    GetOption option = defaultGetOption == null ? GetOption.None : defaultGetOption;
+    Object value = restRequest.getArgs().get(RestUtils.Headers.GET_OPTION);
     if (value != null) {
       String str = (String) value;
       boolean foundMatch = false;
       for (GetOption getOption : GetOption.values()) {
         if (str.equalsIgnoreCase(getOption.name())) {
```

Comment:
It seems inconsistent that this method uses a more relaxed case-insensitive option string matching than the FrontendConfig constructor uses.  Perhaps the FrontEndConfig can do something similar.

Reply:
This is more because http headers are supposed to case insensitive and so some http clients (and servers) turn everything into lower case. You make a good point about consistency but I think the frontend config is more predictable so if the casing isn't exact, it is ok to fail (trying to make it case insensitive is also a lot more code but can be added to `VerifiableProperties` if you think that is required)

Reply:
Ok, that sounds reasonable.